assets/js/modules/reader-revenue-manager/datastore/publications.js

```diff
@@ -53,6 +51,63 @@ const fetchGetPublicationsStore = createFetchStore( {
 	reducerCallback: ( state, publications ) => ( { ...state, publications } ),
 } );
 
+const fetchGetSyncPublicationOnboardingStateStore = createFetchStore( {
+	baseName: 'getSyncPublicationOnboardingState',
+	controlCallback: ( { publicationID, publicationOnboardingState } ) =>
+		API.set(
+			'modules',
+			READER_REVENUE_MANAGER_MODULE_SLUG,
+			'sync-publication-onboarding-state',
+			{
+				publicationID,
+				publicationOnboardingState,
+			}
+		),
+	argsToParams: ( { publicationID, publicationOnboardingState } ) => {
+		return { publicationID, publicationOnboardingState };
+	},
+	validateParams: ( { publicationID, publicationOnboardingState } = {} ) => {
+		invariant(
+			typeof publicationID === 'string' && publicationID.length > 0,
+			'publicationID is required and must be string.'
+		);
+
+		invariant(
+			typeof publicationOnboardingState === 'string' &&
+				publicationOnboardingState.length > 0,
+			'publicationOnboardingState is required and must be string.'
+		);
+	},
+	reducerCallback: (
+		state,
+		{ publicationOnboardingState, isSavedSetting }
+	) => {
+		if ( ! publicationOnboardingState ) {
+			return state;
+		}
+
+		return {
+			...state,
+			settings: {
+				...state.settings,
+				publicationOnboardingState,
+				// eslint-disable-next-line sitekit/no-direct-date
+				publicationOnboardingStateLastSyncedAtMs: Date.now(),
+			},
+			savedSettings: {
+				publicationOnboardingState: isSavedSetting
+					? publicationOnboardingState
+					: state.savedSettings?.publicationOnboardingState,
+				publicationOnboardingStateLastSyncedAtMs: isSavedSetting
+					? // eslint-disable-next-line sitekit/no-direct-date
+					  Date.now()
+					: state.savedSettings
+							?.publicationOnboardingStateLastSyncedAtMs,
+			},
+		};
+	},
```

Comment:
This is implemented pretty much as the IB specifies, however I've realised that it actually introduces a bug and we need to diverge from the IB to avoid it.

At present, on the setup and settings screens, if the user selects a different publication while the sync request is in flight, the new onboarding state will be incorrectly applied to the newly selected publication. I've recorded a screencast to illustrate this:

https://github.com/user-attachments/assets/57f0a79d-eadc-416b-917c-fc5b3438f477

To mitigate this we need to return the `publicationID` in the sync response and use that to determine which state gets updated (and so there's no need to return `isSavedSetting`). We'll also need to update the onboarding state for the publication found in `state.publications`. I'd recommend using `createReducer()` from the `googlesitekit-data` module to use Immer for the reducer so we can update the state directly, simplifying the reducer logic.

```suggestion
	reducerCallback: createReducer(
		( state, { publicationID, publicationOnboardingState } ) => {
			if ( ! publicationID ) {
				return;
			}

			// eslint-disable-next-line sitekit/no-direct-date
			const publicationOnboardingStateLastSyncedAtMs = Date.now();

			if ( state.settings.publicationID === publicationID ) {
				state.settings.publicationOnboardingState =
					publicationOnboardingState;

				state.settings.publicationOnboardingStateLastSyncedAtMs =
					publicationOnboardingStateLastSyncedAtMs;
			}

			if ( state.savedSettings.publicationID === publicationID ) {
				state.savedSettings.publicationOnboardingState =
					publicationOnboardingState;

				state.savedSettings.publicationOnboardingStateLastSyncedAtMs =
					publicationOnboardingStateLastSyncedAtMs;
			}

			const publication = state.publications.find(
				// eslint-disable-next-line sitekit/acronym-case
				( { publicationId: id } ) => id === publicationID
			);

			if ( publication ) {
				publication.onboardingState = publicationOnboardingState;
			}
		}
	),
```

Cc @nfmohit